### ISSUE TITLE
AuthInfo.getSfdxAuthUrl method returns wrong Authorization URL

### ISSUE THREAD

**Issue by saurabh-deep** [forcedotcom/sfdx-core#106]
### Summary

AuthInfo.getSfdxAuthUrl method returns wrong Authorization URL

### Steps To Reproduce:

1.  Authorize an Org using Salesforce CLI => **sfdx force:auth:web:login**
1.  Execute **sfdx force:org:display -u above@username --verbose** command to see the SFDX Auth URL of the org authorized in the first step.
1. Create a small NodeJS program like the following to use **@salesforce/core** for extracting the SFDX Auth URL from the already authorized org above.

``` JavaScript
const { AuthInfo } = require('@salesforce/core');

AuthInfo.create({ username: 'above@username' })
  .then((authInfoObj) => {
    console.log('Auth URL >>>', authInfoObj.getSfdxAuthUrl());
  })
  .catch(err => {
    console.log('Error :::', err);
  });
```

### Expected result

**sfdx force:org:display** command in step 2 and the Node JS program using **@salesforce/core** in step 3 should produce the same SFDX Auth URL.

### Actual result

While the AuthURL displayed by **sfdx force:org:display** is correct, the one extracted using **@salesfore/core** module is incorrect. In place of client secret, it is inserting string `undefined` after the client ID. 

Therefore, if the Auth URL displayed by sfdx force:org:display is - `force://PlatformCLI::something.somethingmore@incredible-astro-xxxxxx.my.salesforce.com` 

Then the same produced using @salesforce/core is - `force://PlatformCLI:undefined:something.somethingmore@incredible-astro-xxxxxx.my.salesforce.com`

### Additional information

The problem seems to be in the getSfdxAuthUrl logic implemented [here](https://github.com/forcedotcom/sfdx-core/blob/develop/src/authInfo.ts#L554). There is no check if the decryptedFields.clientSecret is falsy or undefined. 

``` JavaScript
if (decryptedFields.clientId) {
  sfdxAuthUrl += `${decryptedFields.clientId}:${decryptedFields.clientSecret}:`;
}
sfdxAuthUrl += `${decryptedFields.refreshToken}@${instanceUrl}`;
```

`salesforce-alm` module checks that and in case the clientSecret is falsy then it replaces the same with an empty string. Following is the corresponding line from salesforce-alm module in `lib/orgDescribeApi.js` file. The same or similar needs to be done in `@salesforce/core` as well to get the same Auth URL as returned by the Salesforce CLI.

``` JavaScript
const clientSecret = oauth2.clientSecret || '';
orgDesc.sfdxAuthUrl = `force://${clientId}:${clientSecret}:${refreshToken}@${instanceUrl}`;
```

**@salesforce/core Version** : 1.2.2

**SFDX CLI Version** : 6.56.0

**SFDX plugin Version** : 45.5.0

**OS and version**: macOS High Sierra (10.13.6)

**Comment by clairebianchi** [comment]
Thank you for reporting this issue, we have added it to our backlog. I will report back once it is fixed.

**Comment by amphro** [comment]
This was fixed in #148
